packages/plugin-core/src/commands/VaultAddCommand.ts

```diff
@@ -182,8 +259,85 @@ export class VaultAddCommand extends BasicCommand<CommandOpts, CommandOutput> {
     return { vaults, workspace };
   }
 
+  async handleRemoteRepoSelfContained(
+    opts: CommandOpts
+  ): Promise<{ vaults: DVault[] }> {
+    return window.withProgress(
+      {
+        location: ProgressLocation.Notification,
+        title: "Adding remote vault",
+        cancellable: false,
+      },
+      async (progress) => {
+        const { wsRoot } = ExtensionProvider.getDWorkspace();
+        progress.report({
+          message: "cloning repo",
+          increment: 0,
+        });
+        const { name, pathRemote: remoteUrl } = opts;
+        const localUrl = path.join(wsRoot, opts.path);
+        if (!remoteUrl) {
+          throw new DendronError({
+            message:
+              "Remote vault has no remote set. This should never happen, please send a bug report if you encounter this.",
+          });
+        }
+
+        await fs.ensureDir(localUrl);
+        const git = new Git({ localUrl, remoteUrl });
+        // `.` so it clones into the `localUrl` directory, not into a subdirectory of that
+        await git.clone(".");
+        const { vaults, workspace } = await GitUtils.getVaultsFromRepo({
+          repoPath: localUrl,
+          wsRoot,
+          repoUrl: remoteUrl,
+        });
+        if (_.size(vaults) === 1 && name) {
+          vaults[0].name = name;
+        }
+        // add all vaults
+        const increment = 100 / (vaults.length + 1);
+        progress.report({
+          message:
+            vaults.length === 1
+              ? "adding vault"
+              : `adding ${vaults.length} vaults`,
+          increment,
+        });
+        const wsService = new WorkspaceService({ wsRoot });
+
+        if (workspace) {
+          // TODO: To be implemented for backwards compatibility until workspace vaults are phased out
+          throw new DendronError({
```

Comment:
is it worth adding compatibility to workspace vaults? another approach is to, when we convert from workspace to self contained vaults, to also convert over existing workspace vaults to self contained vaults.
this also reminds me - what is the migration plan for the dendron team? how will we dogfood self contained vaults with `org-workspace`?

Reply:
My preference here would be to add support for them, because adding support for these is pretty similar to supporting the transitive dependencies for self contained vaults.

> migration plan for the dendron team

My plan was to convert one vault (probably `org-private`, or something else that's private to us) manually to begin with once this PR hits nightly (or general release if we have members who don't use nightly)

Once we have a general release out that has this PR + the ability to open self contained vaults by themselves (which is what I'll be working on next), then I'll go ahead and convert the remaining Dendron vaults as well (hopefully with an automatic convert if I have that implemented by then, manually otherwise)